common/PersistenceFormat.js

```diff
@@ -61,52 +67,151 @@ class PersistenceFormatBase {
             });
     }
 
-    getEntryHash (entryId) {
+    getEntryHash(entryId) {
         return entryId.charAt(entryId.length - 1);
     }
-}
 
+    getFlavorHash() {
+        let hours = new Date().getHours().toString();
+        return hours < 10 ? ("0" + hours) : hours;
+    }
 
+    getBasePathFromFull(fullPath) {
+        // cut away both flavor and time components
+        let lastSepIdx = _.lastIndexOf(fullPath, path.sep) - 1;
+        return fullPath.substring(0, _.lastIndexOf(fullPath, path.sep, lastSepIdx) + 1)
+    }
 
-if(!preserveOriginalHLS) {
-    class DefaultPersistenceFormat extends PersistenceFormatBase {
+    getFlavorPath(destPath, param) {
+        let hours = new Date().getHours().toString();
+        let hash = hours < 10 ? ("0" + hours) : hours;
+        let fullPath = path.join(destPath, hash);
+        return {fullPath, hash};
+    }
 
-        getBasePathFromFull(fullPath) {
-            // cut away both flavor and time components
-            let lastSepIdx = _.lastIndexOf(fullPath,path.sep) - 1;
-            return fullPath.substring(0,_.lastIndexOf(fullPath,path.sep,lastSepIdx)+1)
+    compressChunkName(tsChunkName) {
+        var matched = tsChunktMatch.exec(tsChunkName);
+        if (matched) {
+            return matched[1] + '-' + matched[2] + '.mp4';
         }
+        return tsChunkName;
+    }
 
-        getFlavorPath (destPath,param) {
-           let hours = new Date().getHours().toString();
-            let hash = hours < 10 ? ("0" + hours) : hours;
-            let fullPath = path.join(destPath, hash);
-            return { fullPath, hash };
+    compressTsChunkName(tsChunkName) {
+        var matched = tsChunktMatch.exec(tsChunkName);
+        if (matched) {
+            return matched[1] + '-' + matched[2] + '.ts';
         }
+        return tsChunkName;
+    }
+
+
+}
+
 
-        compressChunkName(tsChunkName) {
-            var matched = tsChunktMatch.exec(tsChunkName);
-            if (matched) {
-                return matched[1] + '-' + matched[2] + '.mp4';
+if (!preserveOriginalHLS.enable) {
```

Comment:
it looks like checking for dirHierarchy && preserveOriginalHLS.enable will simplify the code